nifi-extension-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/GetS3ObjectMetadata.java

```diff
@@ -0,0 +1,208 @@
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.components.Validator;
+import org.apache.nifi.expression.ExpressionLanguageScope;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static org.apache.nifi.processors.aws.util.RegionUtilV1.S3_REGION;
+
+@Tags({"Amazon", "S3", "AWS", "Archive", "Exists"})
+@InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
+@CapabilityDescription("Check for the existence of a file in S3 without attempting to download it. This processor can be " +
+        "used as a router for work flows that need to check on a file in S3 before proceeding with data processing")
+@SeeAlso({PutS3Object.class, DeleteS3Object.class, ListS3.class, TagS3Object.class, DeleteS3Object.class, FetchS3Object.class})
+public class GetS3ObjectMetadata extends AbstractS3Processor {
+    public static final AllowableValue TARGET_ATTRIBUTES = new AllowableValue("attributes", "Attributes", "When " +
+            "selected, the metadata will be written to FlowFile attributes that have a user-configured prefix. For example: " +
+            "the standard S3 attribute Content-Type will be written as s3.Content-Type when using the default value. User-defined metadata " +
+            "will be included in the attributes added to the FlowFile");
+    public static final AllowableValue TARGET_FLOWFILE_BODY = new AllowableValue("flowfile-content", "FlowFile Body", "Write " +
+            "the metadata to the FlowFile's content as JSON data.");
+
+    public static final PropertyDescriptor METADATA_TARGET = new PropertyDescriptor.Builder()
+            .name("Metadata Target")
+            .description("This determines where the metadata will be written when it is found.")
+            .addValidator(Validator.VALID)
+            .required(true)
+            .allowableValues(TARGET_ATTRIBUTES, TARGET_FLOWFILE_BODY)
+            .defaultValue(TARGET_ATTRIBUTES)
+            .build();
+
+    public static final PropertyDescriptor METADATA_ATTRIBUTE_PREFIX = new PropertyDescriptor.Builder()
+            .name("Metadata Attribute Prefix")
+            .description("The prefix for FlowFile attributes generated from the S3 object metadata.")
+            .addValidator(StandardValidators.NON_EMPTY_EL_VALIDATOR)
+            .defaultValue("s3.")
+            .dependsOn(METADATA_TARGET, TARGET_ATTRIBUTES)
+            .required(true)
+            .build();
```

Comment:
Based on the fact that `ListS3` always uses the `s3.` prefix, this seems unnecessary as a configurable property.

Reply:
Done.